lib/common/socketpool.c

```diff
@@ -47,13 +48,18 @@ struct st_h2o_socketpool_connect_request_t {
     h2o_loop_t *loop;
     h2o_hostinfo_getaddr_req_t *getaddr_req;
     h2o_socket_t *sock;
-    struct {
-        h2o_multithread_receiver_t *getaddr_receiver;
-        h2o_socketpool_target_vector_t *targets;
-        size_t selected;
-        int *tried;
-        size_t try_count;
-    } lb;
+    h2o_multithread_receiver_t *getaddr_receiver;
+    union {
+        struct {
+            h2o_socketpool_target_t *target;
+        } single;
+        struct {
+            h2o_socketpool_target_vector_t *targets;
+            size_t selected;
+            int *tried;
+            size_t try_count;
+        } lb;
+    };
```

Comment:
Can't this be changed to something like:
```
h2o_multithread_receiver_t *getaddr_receiver;
size_t selected_target;
struct {
    int *tried;
    size_t try_count;
} lb;
```

What I am suggesting here is that you only need to retain the selected index of that designates the target within `pool->targets`. I believe that not using unions would reduce the impact on the code-base.

Note that the proposed struct removes `lb.targets`. I believe that it is possible; please see 5780763.